### ISSUE TITLE
Handle destructuring in 'object-literal-shorthand'?

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3272]
### Feature request / Bug report / Discussion

The current implementation of `object-literal-shorthand` only checks properties of object literals.
Destructuring assignments are also checked:
```ts
// the code doesn't make much sense, but that's not the point
// there's an error in both object literals - that's expected
({foo: foo} = {foo: foo});
```

And then there are variable or parameter declarations using destructuring. They look pretty much the same. But they are currently not checked by the rule.

```ts
let {foo: foo} = obj; // I'd expect an error here
```

Is there a reason for the current behavior or was it just forgotten?

IMO adding this additional check qualifies as bugfix.

**Comment by rjmunro** [comment]
I have the opposite problem; in general I don't want shorthand. I have the TSLint config `"object-literal-shorthand": [true, "never"]` but in the case of a destructuring assignment, I don't want to double everything:

```
({foo, bar} = {foo: 'string 1', bar: 'string 2'});
```

I think we need to control destructuring separately from normal object-literal-shorthand.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
